volatility3/framework/layers/intel.py

```diff
@@ -259,6 +264,10 @@ def is_valid(self, offset: int, length: int = 1) -> bool:
         except exceptions.InvalidAddressException:
             return False
 
+    def is_dirty(self, offset: int) -> bool:
```

Comment:
My slight concern is that we only add this to the intel layer, and not the underlying `TranslationLayerInterface`, but I think we've done that in other places.  I guess it just requires that any time we use it we ensure we know we're working with an intel layer (at the moment, that's likely to be all we're working with, but in the future it could be an arm layer or something else)...